### PR TITLE
Implementation of features of usim 0.3.0

### LINKED ISSUE #20: Adapt simulation to current usim features
The latest version of usim introduces some useful features such as the borrowing of resources that exactly match the use case we require for lapis. Therefore, this should be adapted.

### PR DESCRIPTION
**Pull request by eileen-kuehn**
The release of usim 0.3.0 allows the borrowing/claiming of resources. This simplified the code a lot and further helps guaranteeing the proper usage of available resources.
Further the scheduling now also takes into account the limits of the given resource containers of usim.

Closes #20.

### CODE REVIEW

File: lapis/drone.py [hunk 1]
```diff
@@ -47,9 +48,16 @@ def __init__(self, scheduler, pool_resources: dict,
...
-            key: self.pool_resources[key] - self.resources[key]
-            for key in self.pool_resources
+            key: getattr(levels, key) for key in self.pool_resources
+        }
+
+    @property
+    def available_resources(self):
+        levels = self.used_resources.levels
+        return {
+            key: getattr(levels, key) for key in self.pool_resources
```
Comment (maxfischer2781) [inline lapis/drone.py]: You may want to exploit that ``ResourceLevels`` are an iterable over ``field, value`` pairs. Try if ``dict(self.used_resources.levels)`` works for you.

File: lapis/job.py [hunk 2]
```diff
@@ -27,6 +27,12 @@ def __init__(self, resources: dict, used_resources: dict, in_queue_since: float
         """
         self.resources = resources
         self.used_resources = used_resources
+        for key in used_resources:
+            if key not in resources:
+                logging.getLogger("implementation")\
+                    .info("job uses different resources than specified, added",
```
Comment (maxfischer2781) [inline lapis/job.py]: If you put the ``.info(`` on the ``logging.GetLogger ...`` line, then you can put the arguments on the next lines without a literal line break.

File: lapis/scheduler.py [hunk 3]
```diff
@@ -101,23 +101,23 @@ def _schedule_job(self, job) -> Drone:
...
+                    # and all resources that are too small to even be considered
                     cost = float("Inf")
                     break
                 else:
-                    cost += (drone.pool_resources[resource_type]
-                             - drone.resources[resource_type]) // \
-                        job.resources[resource_type]
-            cost /= len(resource_types)
+                    try:
+                        cost += 1 / (resources.get(resource_type, 0)
```
Comment (maxfischer2781) [inline lapis/scheduler.py]: Doesn't a default of 0 cause a ``ZeroDivisionError``? You may want to have no default and fail faster with a ``KeyError``.

File: lapis/drone.py (hunk 1 above)
Comment (eileen-kuehn) [reply lapis/drone.py]: Seems to work, thanks.

File: lapis/job.py (hunk 2 above)
Comment (eileen-kuehn) [reply lapis/job.py]: Just created #24 that will take care on this.

File: lapis/scheduler.py (hunk 3 above)
Comment (eileen-kuehn) [reply lapis/scheduler.py]: Thanks, took care on this.

**Pull request merged**